helpdesk/teams/views.py

```diff
@@ -172,3 +173,49 @@ def get_entries(self) -> QuerySet[Any]:
 
     def get_context_data(self, **kwargs: Any) -> dict[str, Any]:
         return super().get_context_data(entries=self.get_entries(), **kwargs)
+
+
+class TeamAttendanceView(LoginRequiredMixin, SingleTableMixin, ListView):
+    model = Team
+    table_class = TeamAttendanceTable
+
+    def get_queryset(self) -> QuerySet[Any]:
+        return Team.objects.all().prefetch_related(
+            Prefetch(
+                "team_attendance_events",
+                TeamAttendanceEvent.objects.order_by("-created_at")[:1],
+                to_attr="latest_event",
+            )
+        )
+
+
+class TeamAttendanceFormView(LoginRequiredMixin, CreateView):
+    http_method_names = ["get", "post"]
+    model = TeamAttendanceEvent
+    form_class = TeamAttendanceLogForm
+
+    def get_initial(self) -> dict[str, Any]:
+        return {
+            "team": get_object_or_none(Team, tla=self.kwargs["tla"]),
+        }
+
+    def get_success_url(self) -> str:
+        return reverse_lazy("teams:team_list_attendance")
+
+    def form_valid(self, form: TeamAttendanceLogForm) -> HttpResponse:
+        assert self.request.user.is_authenticated
+        team = form.cleaned_data["team"]
+        team.team_attendance_events.create(
```

Comment:
Issue: You should be able to use `form.save()` to do this instead. 

https://forum.djangoproject.com/t/modify-form-field-before-saving-it/4090/3 is a good example.